### ISSUE TITLE
[Question] maven authentication with hostrules changed?

### ISSUE THREAD

**Issue by 123Haynes** [renovatebot/config-help#241]
### Which Renovate are you using? CLI, App, or Pro

The latest docker image, so i guess CLI

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps

Selfhosted Gitlab

### Have you checked the logs? Don't forget to include them if relevant
config.js:  

```
module.exports = {
	platform: 'gitlab',
	endpoint: 'https://git.private/api/v4/',
	labels: ['renovate'],
	logLevel: 'info',
	maven: {
	    "enabled": true
	},
	"packageRules": [{
        "managers": ["maven"],
	    "registryUrls": ["https://artifactory.private/artifactory/ressource-maven-central/", "https://artifactory.private/artifactory/ressource-maven-oracle/"]
    },
    {
        "managers": ["docker"],
	    "registryUrls": ["https://ressource-docker-hub.artifactory.private"]
    }],
	hostRules: [{
	    "platform": "maven",
	    "endpoint": "https://artifactory.private/artifactory/ressource-maven-central/",
	    "username": "artifactoryusername",
	    "password": "artifactorypassword"
	},
	{
	    "platform": "maven",
	    "endpoint": "https://artifactory.private/artifactory/ressource-maven-oracle/",
	    "username": "artifactoryusername",
	    "password": "artifactorypassword"
	},
	{
	    "platform": "docker",
	    "endpoint": "https://ressource-docker-hub.artifactory.private",
	    "username": "artifactoryusername",
	    "password": "artifactorypassword"
	}],
	onboardingConfig: { extends: ["config:base"] }
};
```

```
WARN: Dependency lookup unauthorized. Please add authentication with a hostRule (repository=privaterepo)
       "pkgUrl": "https://artifactory.private/artifactory/ressource-maven-central/org/springframework/boot/spring-boot-starter-parent/maven-metadata.xml"
DEBUG: org.springframework.boot:spring-boot-starter-parent not found in repository https://artifactory.private/artifactory/ressource-maven-central/ (repository=privaterepo)
DEBUG: Looking up org.springframework.boot:spring-boot-starter-parent in repository #1 - https://artifactory.private/artifactory/ressource-maven-oracle/ (repository=privaterepo)
```

### What is your question?

This config stopped working sometime in the last 2 weeks (last MR created by renovatebot on my repos).  

Did something change on how I have to configure my hostRules so the authentication works?

**Comment by 123Haynes** [comment]
Looking through my CI logs, it looks like this Error started to appear on May 13.  

edit: I changed the version of the docker image to 17.2.3 and renovate works again.
So I guess it broke sometime between 17.2.3 and 17.3.1.  
I'll try to narrow it down more tomorrow.

**Comment by rarkins** [comment]
Thanks for help in narrowing it down

**Comment by rarkins** [comment]
Are you sure that 17.3.1 is a broken version? Latest is 17.11.1

**Comment by rarkins** [comment]
[datasource/maven](https://github.com/renovatebot/renovate/blob/master/lib/datasource/maven/index.js) was last changed 26 days ago, so probably isn't the culprit.

[util/got/host-rules](https://github.com/renovatebot/renovate/commits/master/lib/util/got/host-rules.js) has only had one minor change.

[util/host-rules](https://github.com/renovatebot/renovate/commits/master/lib/util/host-rules.ts) has had quite a few changes though, so is the likely culprit.

**Comment by rarkins** [comment]
One interesting thing I note with your configuration is that you have the same host ("artifactory.private") and two different hostrules for different paths. I didn't think this scenario has ever worked actually, I wonder if somehow it worked by accident. Internally we store hostRules by host but I've been intending to update it to support full endpoints with paths.

**Comment by 123Haynes** [comment]
>Are you sure that 17.3.1 is a broken version? Latest is 17.11.1

No. That's just what I assumed from the date of the releases. But from my test so far today, it looks like something weird is happening....

I tested version 17.4.0 twice. The first time it worked like a charm, the second time it displayed this error.
So either the hostrules are not always being applied correctly for some reason, or the error message has a to broad scope and doesn't catch what's actually happening.  
Maybe it's used for timeouts as well?

I'll try to change the config to one hostrule for the 2 paths now and see if that changes something.

**Comment by rarkins** [comment]
Sounds like something is definitely wrong, but it's certainly weird. Any more help you can provide in tracking down the issue would be appreciated.

**Comment by 123Haynes** [comment]
of course. I'm currently waiting to get access to the artifactory logs, to see what's happening on that side. I'll comment here as soon as I know more.

**Comment by 123Haynes** [comment]
@rarkins so apparently I locked my artifactory account while trying to narrow down the versions.  
That's probably why 17.4.0 worked once and then failed later.  

So one issue is that the error message does not reflect that the access has been denied.
I'm currently waiting for our artifactory admin to unlock my account to narrow the original issue further down.

**Comment by rarkins** [comment]
@123Haynes IIRC, our logic for that message is "Is this a private registry? Did we get an auth error? If so then tell them to add auth". You are right that it's misleading if auth *is* present but is incorrect or locked

**Comment by 123Haynes** [comment]
@rarkins Sooo... I finally know what happend.  
6 Days ago there was an issue with our artifactory instance that invalidated all access tokens.
when I came back from vacation yesterday I changed the access token for most repos, but it looks like I missed one. Because I only run renovate every few hours, my account was already locked by the time renovate tried to access the artifactory.  

So in the end nothing broke, it's just the misleading error message.
Sorry for wasting your time on this :cry:
